query/src/sql/planner/plans/create_table_v2.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+
+use common_datavalues::DataSchemaRef;
+use common_meta_app::schema::CreateTableReq;
+use common_meta_app::schema::TableMeta;
+use common_meta_app::schema::TableNameIdent;
+
+use crate::sql::plans::Plan;
+
+pub type TableOptions = BTreeMap<String, String>;
+
+// Replace `PlanNode` with `Plan`
+#[derive(serde::Serialize, serde::Deserialize, Clone)]
+pub struct CreateTablePlanV2 {
+    pub if_not_exists: bool,
+    pub tenant: String,
+    /// The catalog name
+    pub catalog: String,
+    pub database: String,
+    /// The table name
+    pub table: String,
+
+    pub table_meta: TableMeta,
+
+    pub cluster_keys: Vec<String>,
+    #[serde(skip)]
+    pub as_select: Option<Box<Plan>>,
```

Comment:
Maybe you can use a optimized `PhysicalPlan` here, which is serializable and can be compiled into `Pipeline` easily.